cmd/x-check-mailserver/work.go

```diff
@@ -40,163 +43,105 @@ func NewWorkUnit(mailEnode string, config *params.NodeConfig) *WorkUnit {
 
 // WorkUnitConfig configures the execution of the work.
 type WorkUnitConfig struct {
-	From     uint32
-	To       uint32
-	Channels []string
+	From  uint32
+	To    uint32
+	Chats []string
 }
 
 // Execute runs the work.
-func (u *WorkUnit) Execute(config WorkUnitConfig, mailSignals *signalForwarder) error {
+func (u *WorkUnit) Execute(config WorkUnitConfig) error {
 	if err := u.startNode(); err != nil {
 		return fmt.Errorf("failed to start node: %v", err)
 	}
 
-	if err := u.addPeer(); err != nil {
-		return fmt.Errorf("failed to add peer: %v", err)
+	if err := u.startMessenger(); err != nil {
+		return fmt.Errorf("failed to create messenger: %v", err)
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-	defer cancel()
-	mailServerSymKeyID, err := u.shh.GenerateSymmetricKeyFromPassword(
-		ctx, protocol.MailServerPassword)
-	if err != nil {
-		return fmt.Errorf("failed to generate sym key for mail server: %v", err)
+	if err := u.addPeer(u.MailServerEnode); err != nil {
+		return fmt.Errorf("failed to add peer: %v", err)
 	}
 
-	var topics []whisper.TopicType
-	for _, ch := range config.Channels {
-		topic, err := protocol.PublicChatTopic([]byte(ch))
-		if err != nil {
-			return fmt.Errorf("failed to create topic: %v", err)
+	for _, chatName := range config.Chats {
+		chat := protocol.CreatePublicChat(chatName, u.messenger.Timesource())
+		if err := u.messenger.Join(chat); err != nil {
+			return err
 		}
-		topics = append(topics, topic)
-	}
-
-	var messageSubErrs []<-chan error
-	messages := make(chan *whisper.Message)
-
-	for _, ch := range config.Channels {
-		symKeyID, err := addPublicChatSymKey(u.shh, ch)
-		if err != nil {
-			return fmt.Errorf("failed to add sym key for channel '%s': %v", ch, err)
+		if err := u.messenger.SaveChat(&chat); err != nil {
+			return err
 		}
+	}
 
-		sub, err := subscribeMessages(u.shh, ch, symKeyID, messages)
-		if err != nil {
-			return fmt.Errorf("failed to subscribe for messages: %v", err)
-		}
-		defer sub.Unsubscribe()
-		messageSubErrs = append(messageSubErrs, sub.Err())
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
Why a second? Same reason as in the other case?

Reply:
No, but it should not take more than 1 second to send a request to a mailserver. It's arbitrary number just like you would make an HTTP request and want to return if it takes too long.